### PR TITLE
docs: specify the ante handler

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2044

### CODE REVIEW

**Review by evan-forbes — DISMISSED** [review]
nice!

File: app/ante/ante.go [hunk 1]
```diff
@@ -21,24 +21,46 @@ func NewAnteHandler(
...
+		// Contract: must be called before all signature verification decorators.
+		ante.NewSetPubKeyDecorator(accountKeeper),
+		// Ensure that the tx's count of signatures is <= the tx signature limit.
 		ante.NewValidateSigCountDecorator(accountKeeper),
+		// Ensure that the tx's gas limit is > the gas consumed based on signature verification.
+		// Side effect: consumes gas from the gas meter.
 		ante.NewSigGasConsumeDecorator(accountKeeper, sigGasConsumer),
+		// Ensure that the tx's signatures are valid.
+		// Note: does not consume gas from the gas meter.
 		ante.NewSigVerificationDecorator(accountKeeper, signModeHandler),
```
Comment (cmwaters) [inline app/ante/ante.go:51]: This also checks the nonces (sequence number)

File: specs/src/specs/ante_handler.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# AnteHandler
+
+Celestia makes use of a Cosmos SDK [AnteHandler](https://docs.cosmos.network/v0.46/modules/auth/03_antehandlers.html#antehandlers) in order to reject decodable sdk.Txs that do not meet certain criteria. The ante handler is defined in [app/ante/ante.go](https://github.com/celestiaorg/celestia-app/blob/7f97788a64af7fe0fce00959753d6dd81663e98f/app/ante/ante.go). It chains together several decorators to ensure the following criteria are met:
```
Comment (cmwaters) [inline specs/src/specs/ante_handler.md:3]: Can you add (somewhere appropriate) that it's used in four areas: `CheckTx` when a transaction goes into the mempoool, `PrepareProposal` for when building a block, `ProcessProposal`, for validating blocks and `DeliverTx`, for executing the transaction

File: specs/src/specs/block_validity_rules.md [hunk 3]
```diff
@@ -41,7 +41,9 @@ Celestia specifc validity rules can be categorized into two groups:
 
 All `BlobTx` transactions must be valid according to the [BlobTx validity rules](../../../x/blob/README.md#validity-rules)
 
-All remaining transaction types do not have to by valid if included in a block. For a complete list of modules see [state machine modules](./state_machine_modules.md).
+All remaining transaction types do not have to be valid if included in a block. In other words: random tx data is permitted, but decodable sdk.Txs must pass all [AnteHandler](./ante_handler.md) checks in order for the block to be valid.
```
Comment (cmwaters) [inline specs/src/specs/block_validity_rules.md:44]: Note that in `CheckTx`, we actually try to decode the txs as sdk txs else we error:

https://github.com/celestiaorg/celestia-app/blob/cbe5020e35d17bb26da298b9bb14c16ac4fb06ba/app/check_tx.go#L21

**Review by cmwaters — APPROVED** [review]
Looks good. Just a few comments from me

File: app/ante/ante.go (hunk 1 above)
Comment (rootulp) [reply app/ante/ante.go]: Good call! Described in da09fa2766886e8c7e514b7ed95529771a62861f

File: specs/src/specs/ante_handler.md (hunk 2 above)
Comment (rootulp) [reply specs/src/specs/ante_handler.md]: Good idea! Resolved in 223a80f1c93c5e3e44b3ecfbfcb01262356c6625

File: specs/src/specs/block_validity_rules.md (hunk 3 above)
Comment (rootulp) [reply specs/src/specs/block_validity_rules.md]: Good catch! Resolved via 6e54aaa0ee5027cd04b8c845cf8d4eeacc2d8340

**Review by evan-forbes — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]

**Pull request merged**